src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -345,7 +365,11 @@ public String getFullName() {
      * @since 1.515
      */
     public String getFullDisplayName() {
-    	return TestNameTransformer.getTransformedName(getFullName());
+        StringBuilder transformedName = new StringBuilder(TestNameTransformer.getTransformedName(getFullName()));
```

Comment:
I would prefer

``` 
return this.archiveId == null ? TestNameTransformer.getTransformedName(getFullName()) : TestNameTransformer.getTransformedName(getFullName()) + "[" + this.archiveId + "]";
```

or an equivalent. We should not be allocating a `StringBuilder` for the verbatim case